docs/compactor.md

```diff
@@ -0,0 +1,36 @@
+
+## Solution
+**Compaction command:** To help with the problem mentioned earlier, the PR https://github.com/gardener/etcd-backup-restore/pull/301 introduces compaction command that reduces the size of full snapshot. So, after restoration from the full snapshot, ETCD database is stripped off of old revisions.
+
+**How the PR works:** The compact command added through PR does not disturb the running ETCD while captuing the compacted snapshot. The command is run separately alongside main ETCD process. 
+When the compact command is run, it fetches the latest full snapshot and delta snapshots from the store. Then it restores an embedded etcd server from the snapshots. Then it runs compact and defragment operation on embedded ETCD database. Then it takes the snapshot of the embedded ETCD database and save it into the store.
+
+#### **Points to take care while saving the compacted snapshot:**
+As compacted snapshot and full snapshot are taken by different processes running in different pods but accessing same store to save the snapshots, some problems may arise:
+1.	If there are both full snapshots and compacted snapshots, a pointer is needed to let the restorer know which is what. We can come up with a metadata file that would point to correct snapshot file that can be used to restore.
+2.	If frequency of taking full snapshots is more than taking compacted snapshots, there might be situations when compacted snapshots be totally ignored while restoring as full snapshots might have latest timestamp.
+3.	Uploading the compacted snapshot to the store should be atomic.
+
+Our proposed approach is to remove the responsibility of taking full snapshot from ETCD Backup Restore. Only compactor (a job that will run compact command at regular interval) will take compacted snapshot and save to store. Currently full snapshots and the subsequent delta snapshots are grouped under same prefix. We are generating multiple prefixes named after the timestamps of full snapshots. Our proposal is to flatten the prefixes and save all compacted snapshots and delta snapshots under same prefix. Also, name snapshots after revision number instead of timestamps. Name a delta snapshot after the revision number of first command captured in that particular delta snapshot. Restorer will restore from compacted snapshots that have highest revision number in the name and delta snapshots that have higher revision numbers in name than the comapcted full snapshot.
+
+**Note:** ETCD Backup Restore needs to take one full snapshot just after starting. Compactor will work only if there is any full snapshot already present in the store. It is not limitation but a design choice
```

Comment:
I have already suggested to limit the section above only to the _description_ of the problems arising out of compaction. It might be better to dedicate the rest of the section to the solutions for these problems. It might also be a good idea to split this into two separate sections. 
1. Folder structure and metadata change and rationale behind it. Preferably, with examples for existing and proposed structure/metadata and backward-compatibility. This can be referred to directly in the PR for folder structure change which can be release independent of compaction.
2. How the proposed folder structure/metadata will be used in compaction and what other changes are applicable (no scheduled full snapshot, first full snapshot still being required, backward compatibility for scheduled full snapshot for at least a release or two etc.)